[PyTorch] Enable explicit ATen level sources for lite interpreter

cccclai opened the pull request:
## Summary 

Enable partial explicit Aten level sources list for lite interpreter. More aten level source list will be added.

**Android**
x86:
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh x86`

libpytorch_jni_lite.so -- 3.8 MB

armeabi-v7a
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh armeabi-v7a`
libpytorch_jni_lite.so -- 2.8 MB

**iOS:**
```
(base) chenlai@chenlai-mp install % du -sh *
 15M	include
 58M	lib
2.8M	share
```

```
(base) chenlai@chenlai-mp lib % ls -lh
total 119600
-rw-r--r--  1 chenlai  staff   3.2M Mar  4 23:16 libXNNPACK.a
-rw-r--r--  1 chenlai  staff   910K Mar  4 23:16 libc10.a
-rw-r--r--  1 chenlai  staff   4.6K Mar  4 23:16 libclog.a
-rw-r--r--  1 chenlai  staff    42K Mar  4 23:16 libcpuinfo.a
-rw-r--r--  1 chenlai  staff   1.5M Mar  4 23:16 libeigen_blas.a
-rw-r--r--  1 chenlai  staff    44K Mar  4 23:16 libpthreadpool.a
-rw-r--r--  1 chenlai  staff   166K Mar  4 23:16 libpytorch_qnnpack.a
-rw-r--r--  1 chenlai  staff   384B Mar  4 23:16 libtorch.a
-rw-r--r--  1 chenlai  staff    52M Mar  4 23:16 libtorch_cpu.a
```


### **Master (Baseline):**

**Android:** 
x86: libpytorch_jni_lite.so -- 3.9 MB
armeabi-v7a: libpytorch_jni_lite.so -- 3.0 MB

**IOS:**
`pytorch/build_ios/install/lib` (lib: **57 MB**):
```
(base) chenlai@chenlai-mp lib % ls -lh
total 117744
-rw-r--r--  1 chenlai  staff   3.3M Feb 19 21:28 libXNNPACK.a
-rw-r--r--  1 chenlai  staff   929K Feb 19 21:28 libc10.a
-rw-r--r--  1 chenlai  staff   4.6K Feb 19 21:28 libclog.a
-rw-r--r--  1 chenlai  staff    42K Feb 19 21:28 libcpuinfo.a
-rw-r--r--  1 chenlai  staff   1.5M Feb 19 21:28 libeigen_blas.a
-rw-r--r--  1 chenlai  staff    44K Feb 19 21:28 libpthreadpool.a
-rw-r--r--  1 chenlai  staff   166K Feb 19 21:28 libpytorch_qnnpack.a
-rw-r--r--  1 chenlai  staff   384B Feb 19 21:28 libtorch.a
-rw-r--r--  1 chenlai  staff    51M Feb 19 21:28 libtorch_cpu.a
```
`pytorch/build_ios/install`:
```
(base) chenlai@chenlai-mp install % du -sh *
 15M	include
 57M	lib
2.8M	share
```

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#52769 [PyTorch] Enable explicit ATen level sources for lite interpreter**

Differential Revision: [D26717268](https://our.internmc.facebook.com/intern/diff/D26717268)

cccclai committed:
[PyTorch] Enable explicit ATen level sources for lite interpreter

[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"




[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"




[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"




[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"




[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"




[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"




[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"




[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"




[ghstack-poisoned]

iseeyuan commented on tools/build_variables.bzl:
```diff
@@ -636,7 +636,7 @@ def glob_libtorch_python_sources(gencode_pattern = ":generate-code[{}]"):
     return _libtorch_python_sources
 
-aten_cpu_source_list = [
+aten_source_list_lite = [
```
Could you add comment on what each source list represents?

iseeyuan commented on tools/build_variables.bzl:
```diff
@@ -719,12 +717,55 @@ aten_cpu_source_list = [
     "aten/src/ATen/record_function.cpp",
+]
+
+aten_cpu_source_list = aten_source_list_lite + [
```
Comment

iseeyuan commented on tools/build_variables.bzl:
```diff
@@ -960,3 +964,7 @@ aten_native_source_list = [
+
+# Files in ATen/native with a few exceptions
+# TODO: move the exceptions to proper locations
+aten_native_source_list = aten_fixed_native_source_list + aten_regenereated_native_source_list_lite
```
What's the difference between fixed and regenerated? Could you add comments here?

iseeyuan commented on tools/build_variables.bzl:
```diff
@@ -719,12 +717,55 @@ aten_cpu_source_list = [
+]
+
+aten_cpu_source_list = aten_source_list_lite + [
+    "aten/src/ATen/metal/Context.cpp",
```
Would this be included in lite interpreter? Would metal be needed in iOS GPU?

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"

## Summary 

Enable partial explicit Aten level sources list for lite interpreter. More aten level source list will be added.

x86:
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh x86`

libpytorch_jni_lite.so -- 3.8 MB

armeabi-v7a
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh armeabi-v7a`
libpytorch_jni_lite.so -- 2.8 MB



Differential Revision: [D26717268](https://our.internmc.facebook.com/intern/diff/D26717268)

[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"

## Summary 

Enable partial explicit Aten level sources list for lite interpreter. More aten level source list will be added.

x86:
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh x86`

libpytorch_jni_lite.so -- 3.8 MB

armeabi-v7a
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh armeabi-v7a`
libpytorch_jni_lite.so -- 2.8 MB



Differential Revision: [D26717268](https://our.internmc.facebook.com/intern/diff/D26717268)

[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"

## Summary 

Enable partial explicit Aten level sources list for lite interpreter. More aten level source list will be added.

x86:
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh x86`

libpytorch_jni_lite.so -- 3.8 MB

armeabi-v7a
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh armeabi-v7a`
libpytorch_jni_lite.so -- 2.8 MB



Differential Revision: [D26717268](https://our.internmc.facebook.com/intern/diff/D26717268)

[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"

## Summary 

Enable partial explicit Aten level sources list for lite interpreter. More aten level source list will be added.

x86:
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh x86`

libpytorch_jni_lite.so -- 3.8 MB

armeabi-v7a
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh armeabi-v7a`
libpytorch_jni_lite.so -- 2.8 MB



Differential Revision: [D26717268](https://our.internmc.facebook.com/intern/diff/D26717268)

[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"

## Summary 

Enable partial explicit Aten level sources list for lite interpreter. More aten level source list will be added.

x86:
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh x86`

libpytorch_jni_lite.so -- 3.8 MB

armeabi-v7a
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh armeabi-v7a`
libpytorch_jni_lite.so -- 2.8 MB



Differential Revision: [D26717268](https://our.internmc.facebook.com/intern/diff/D26717268)

[ghstack-poisoned]

iseeyuan reviewed (approved):
LGTM!

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"

## Summary 

Enable partial explicit Aten level sources list for lite interpreter. More aten level source list will be added.

x86:
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh x86`

libpytorch_jni_lite.so -- 3.8 MB

armeabi-v7a
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh armeabi-v7a`
libpytorch_jni_lite.so -- 2.8 MB



Differential Revision: [D26717268](https://our.internmc.facebook.com/intern/diff/D26717268)

[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"

## Summary 

Enable partial explicit Aten level sources list for lite interpreter. More aten level source list will be added.

x86:
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh x86`

libpytorch_jni_lite.so -- 3.8 MB

armeabi-v7a
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh armeabi-v7a`
libpytorch_jni_lite.so -- 2.8 MB



Differential Revision: [D26717268](https://our.internmc.facebook.com/intern/diff/D26717268)

[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"

## Summary 

Enable partial explicit Aten level sources list for lite interpreter. More aten level source list will be added.

x86:
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh x86`

libpytorch_jni_lite.so -- 3.8 MB

armeabi-v7a
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh armeabi-v7a`
libpytorch_jni_lite.so -- 2.8 MB



Differential Revision: [D26717268](https://our.internmc.facebook.com/intern/diff/D26717268)

[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"

## Summary 

Enable partial explicit Aten level sources list for lite interpreter. More aten level source list will be added.

**Android**
x86:
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh x86`

libpytorch_jni_lite.so -- 3.8 MB

armeabi-v7a
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh armeabi-v7a`
libpytorch_jni_lite.so -- 2.8 MB

**iOS:**
```
(base) chenlai@chenlai-mp install % du -sh *
 15M	include
 58M	lib
2.8M	share
```

```
(base) chenlai@chenlai-mp lib % ls -lh
total 119600
-rw-r--r--  1 chenlai  staff   3.2M Mar  4 23:16 libXNNPACK.a
-rw-r--r--  1 chenlai  staff   910K Mar  4 23:16 libc10.a
-rw-r--r--  1 chenlai  staff   4.6K Mar  4 23:16 libclog.a
-rw-r--r--  1 chenlai  staff    42K Mar  4 23:16 libcpuinfo.a
-rw-r--r--  1 chenlai  staff   1.5M Mar  4 23:16 libeigen_blas.a
-rw-r--r--  1 chenlai  staff    44K Mar  4 23:16 libpthreadpool.a
-rw-r--r--  1 chenlai  staff   166K Mar  4 23:16 libpytorch_qnnpack.a
-rw-r--r--  1 chenlai  staff   384B Mar  4 23:16 libtorch.a
-rw-r--r--  1 chenlai  staff    52M Mar  4 23:16 libtorch_cpu.a
```


### **Master (Baseline):**

**Android:** 
x86: libpytorch_jni_lite.so -- 3.9 MB
armeabi-v7a: libpytorch_jni_lite.so -- 3.0 MB

**IOS:**
`pytorch/build_ios/install/lib` (lib: **57 MB**):
```
(base) chenlai@chenlai-mp lib % ls -lh
total 117744
-rw-r--r--  1 chenlai  staff   3.3M Feb 19 21:28 libXNNPACK.a
-rw-r--r--  1 chenlai  staff   929K Feb 19 21:28 libc10.a
-rw-r--r--  1 chenlai  staff   4.6K Feb 19 21:28 libclog.a
-rw-r--r--  1 chenlai  staff    42K Feb 19 21:28 libcpuinfo.a
-rw-r--r--  1 chenlai  staff   1.5M Feb 19 21:28 libeigen_blas.a
-rw-r--r--  1 chenlai  staff    44K Feb 19 21:28 libpthreadpool.a
-rw-r--r--  1 chenlai  staff   166K Feb 19 21:28 libpytorch_qnnpack.a
-rw-r--r--  1 chenlai  staff   384B Feb 19 21:28 libtorch.a
-rw-r--r--  1 chenlai  staff    51M Feb 19 21:28 libtorch_cpu.a
```
`pytorch/build_ios/install`:
```
(base) chenlai@chenlai-mp install % du -sh *
 15M	include
 57M	lib
2.8M	share
```


Differential Revision: [D26717268](https://our.internmc.facebook.com/intern/diff/D26717268)

[ghstack-poisoned]

cccclai committed:
Update on "[PyTorch] Enable explicit ATen level sources for lite interpreter"

## Summary 

Enable partial explicit Aten level sources list for lite interpreter. More aten level source list will be added.

**Android**
x86:
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh x86`

libpytorch_jni_lite.so -- 3.8 MB

armeabi-v7a
`SELECTED_OP_LIST=/Users/chenlai/Documents/pytorch/experiemnt/deeplabv3_scripted.yaml BUILD_LITE_INTERPRETER=1 ./scripts/build_pytorch_android.sh armeabi-v7a`
libpytorch_jni_lite.so -- 2.8 MB

**iOS:**
```
(base) chenlai@chenlai-mp install % du -sh *
 15M	include
 58M	lib
2.8M	share
```

```
(base) chenlai@chenlai-mp lib % ls -lh
total 119600
-rw-r--r--  1 chenlai  staff   3.2M Mar  4 23:16 libXNNPACK.a
-rw-r--r--  1 chenlai  staff   910K Mar  4 23:16 libc10.a
-rw-r--r--  1 chenlai  staff   4.6K Mar  4 23:16 libclog.a
-rw-r--r--  1 chenlai  staff    42K Mar  4 23:16 libcpuinfo.a
-rw-r--r--  1 chenlai  staff   1.5M Mar  4 23:16 libeigen_blas.a
-rw-r--r--  1 chenlai  staff    44K Mar  4 23:16 libpthreadpool.a
-rw-r--r--  1 chenlai  staff   166K Mar  4 23:16 libpytorch_qnnpack.a
-rw-r--r--  1 chenlai  staff   384B Mar  4 23:16 libtorch.a
-rw-r--r--  1 chenlai  staff    52M Mar  4 23:16 libtorch_cpu.a
```


### **Master (Baseline):**

**Android:** 
x86: libpytorch_jni_lite.so -- 3.9 MB
armeabi-v7a: libpytorch_jni_lite.so -- 3.0 MB

**IOS:**
`pytorch/build_ios/install/lib` (lib: **57 MB**):
```
(base) chenlai@chenlai-mp lib % ls -lh
total 117744
-rw-r--r--  1 chenlai  staff   3.3M Feb 19 21:28 libXNNPACK.a
-rw-r--r--  1 chenlai  staff   929K Feb 19 21:28 libc10.a
-rw-r--r--  1 chenlai  staff   4.6K Feb 19 21:28 libclog.a
-rw-r--r--  1 chenlai  staff    42K Feb 19 21:28 libcpuinfo.a
-rw-r--r--  1 chenlai  staff   1.5M Feb 19 21:28 libeigen_blas.a
-rw-r--r--  1 chenlai  staff    44K Feb 19 21:28 libpthreadpool.a
-rw-r--r--  1 chenlai  staff   166K Feb 19 21:28 libpytorch_qnnpack.a
-rw-r--r--  1 chenlai  staff   384B Feb 19 21:28 libtorch.a
-rw-r--r--  1 chenlai  staff    51M Feb 19 21:28 libtorch_cpu.a
```
`pytorch/build_ios/install`:
```
(base) chenlai@chenlai-mp install % du -sh *
 15M	include
 57M	lib
2.8M	share
```


Differential Revision: [D26717268](https://our.internmc.facebook.com/intern/diff/D26717268)

[ghstack-poisoned]
